pkg/option/config.go

```diff
@@ -899,6 +905,14 @@ func (c *DaemonConfig) Populate() {
 	c.Version = viper.GetString(Version)
 	c.Workloads = viper.GetStringSlice(ContainerRuntime)
 
+	// This is a legacy option. Provide backward compatibility by enabling
+	// automatic direct routing. Unlike the old option, it will also enable
+	// direct routing for IPv4. Better than breaking the option. The old
+	// option was not frequently used so this addition in scope is fine.
+	if viper.GetBool(LegacyAutoIPv6NodeRoutesName) {
```

Comment:
Since this is a legacy why does it can overwrite the new option? Shouldn't `c.EnableAutoDirectRouting` default to  `viper.GetBool(LegacyAutoIPv6NodeRoutesName)` and if `viper.GetBool(EnableAutoDirectRoutingName)` is enabled that the option should be overwritten?

Reply:
We want people to *remove* the legacy option in order to turn off the feature, we don't want them to add EnableAutoDirectRoutingName=false in addition. Does that make sense?